zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala

```diff
@@ -383,4 +385,11 @@ class Handlers(mustacheGenerator: ZipkinMustache, queryExtractor: QueryExtractor
           .map(renderTrace(_))
       } getOrElse NotFound
     }
+
+  def handleServices(client: HttpClient, webServiceNameCacheTimeoutInSec: Int): Service[Request, Renderer] =
```

Comment:
actually I think this should move to the query server. You already have a response object, so it is less code. Then all you need to do is forward the request, like we do with `/api/spans`.

```
  get("/api/v1/services") { request: Request =>
    response.ok(spanStore.getAllServiceNames()).cacheControl(servicesMaxAge)
  }
```